### PR TITLE
Initial check-in for Tensorflow Encrypted model execution

### PR DESCRIPTION
**Pull request by vasanth-intel**
Signed-off-by: Vasanth Nagaraja <vasanth.k.nagaraja@intel.com>

### CODE REVIEW

File: baremetal_benchmarking/workloads/tensorflow_workload.py [hunk 1]
```diff
@@ -27,7 +27,7 @@ def download_workload(self, test_config_dict):
         print("\n-- Installing Tensorflow..")
         utils.exec_shell_cmd(TENSORFLOW_INSTALL_CMD)
 
-    def download_bert_models(self):
+    def download_bert_models(self, test_config_dict):
```
Comment (jinengandhi-intel) [inline baremetal_benchmarking/workloads/tensorflow_workload.py:30]: Do we need this new function parameter test_config_dict?

File: baremetal_benchmarking/workloads/tensorflow_workload.py [hunk 2]
```diff
@@ -45,29 +45,43 @@ def download_bert_models(self):
...
+            os.remove("data/" + TF_BERT_FP2_MDOEL_NAME)
+        except OSError:
+            pass
+        print("\n-- Downloading BERT FP32 model..")
+        utils.exec_shell_cmd(TF_BERT_FP32_MODEL_WGET_CMD, None)
+        
+        print("\n-- Required BERT models downloaded..")
 
-    def download_resnet_models(self):
+    def download_resnet_models(self, test_config_dict):
```
Comment (jinengandhi-intel) [inline baremetal_benchmarking/workloads/tensorflow_workload.py:61]: Do we need this new function parameter test_config_dict?

File: baremetal_benchmarking/workloads/tensorflow_workload.py [hunk 3]
```diff
@@ -45,29 +45,43 @@ def download_bert_models(self):
             print("\n-- Downloading BERT checkpoints models..")
             utils.exec_shell_cmd(TF_BERT_CHECKPOINTS_WGET_CMD, None)
             utils.exec_shell_cmd(TF_BERT_CHECKPOINTS_UNZIP_CMD, None)
-            utils.exec_shell_cmd(TF_BERT_FP32_MODEL_WGET_CMD, None)
 
-        print("\n-- Required BERT models downloaded / already present..")
+        # We are deleting the model to avoid complication (with encryption in picture) on whether
```
Comment (jinengandhi-intel) [inline baremetal_benchmarking/workloads/tensorflow_workload.py:49]: We may not need to delete the base model, as its the same version.

File: baremetal_benchmarking/workloads/tensorflow_workload.py [hunk 4]
```diff
@@ -45,29 +45,43 @@ def download_bert_models(self):
...
+    def download_resnet_models(self, test_config_dict):
         if not os.path.exists('./models'):
             print("\n-- Downloading RESNET Intel_AI models..")
             utils.exec_shell_cmd(TF_RESNET_INTEL_AI_MODELS_CLONE_CMD, None)
 
-        resnet_inte8_model_name = os.path.basename(TF_RESNET_INT8_MODEL_WGET_CMD.split()[1])
-        if not os.path.exists(resnet_inte8_model_name):
-            print("\n-- Downloading RESNET Pretrained model..")
-            utils.exec_shell_cmd(TF_RESNET_INT8_MODEL_WGET_CMD, None)
+        # We are deleting the model to avoid complication (with encryption in picture) on whether
```
Comment (jinengandhi-intel) [inline baremetal_benchmarking/workloads/tensorflow_workload.py:66]: We may not need to delete the base model, as its the same version.

File: baremetal_benchmarking/workloads/tensorflow_workload.py [hunk 5]
```diff
@@ -45,29 +45,43 @@ def download_bert_models(self):
...
             utils.exec_shell_cmd(TF_BERT_CHECKPOINTS_WGET_CMD, None)
             utils.exec_shell_cmd(TF_BERT_CHECKPOINTS_UNZIP_CMD, None)
-            utils.exec_shell_cmd(TF_BERT_FP32_MODEL_WGET_CMD, None)
 
-        print("\n-- Required BERT models downloaded / already present..")
+        # We are deleting the model to avoid complication (with encryption in picture) on whether
+        # the model downloaded/present with previous framework execution would be plain or encrypted model.
+        # So, we download fresh model everytime we execute the test.
+        try:
+            os.remove("data/" + TF_BERT_FP2_MDOEL_NAME)
```
Comment (jinengandhi-intel) [inline baremetal_benchmarking/workloads/tensorflow_workload.py:53]: TF_BERT_FP2_MDOEL_NAME typo

**Review by jinengandhi-intel — CHANGES_REQUESTED** [review]

File: baremetal_benchmarking/workloads/tensorflow_workload.py (hunk 1 above)
Comment (vasanth-intel) [reply baremetal_benchmarking/workloads/tensorflow_workload.py]: Done

File: baremetal_benchmarking/workloads/tensorflow_workload.py (hunk 2 above)
Comment (vasanth-intel) [reply baremetal_benchmarking/workloads/tensorflow_workload.py]: Done

File: baremetal_benchmarking/workloads/tensorflow_workload.py (hunk 3 above)
Comment (vasanth-intel) [reply baremetal_benchmarking/workloads/tensorflow_workload.py]: Done. Earlier, was thinking to use the same path for encryption as well. But, then after changing the paths for plain and encrypted models, forgot to remove this snippet.

File: baremetal_benchmarking/workloads/tensorflow_workload.py (hunk 4 above)
Comment (vasanth-intel) [reply baremetal_benchmarking/workloads/tensorflow_workload.py]: Done. Earlier, was thinking to use the same path for encryption as well. But, then after changing the paths for plain and encrypted models, forgot to remove this snippet.

File: baremetal_benchmarking/workloads/tensorflow_workload.py (hunk 5 above)
Comment (vasanth-intel) [reply baremetal_benchmarking/workloads/tensorflow_workload.py]: Done.

**Review by jinengandhi-intel — CHANGES_REQUESTED** [review]

File: baremetal_benchmarking/workloads/tensorflow_workload.py [hunk 6]
```diff
@@ -45,29 +45,29 @@ def download_bert_models(self):
...
 
-        print("\n-- Required RESNET models downloaded / already present..")
+        print("\n-- Required RESNET models downloaded..")
 
     def build_and_install_workload(self, test_config_dict):
         print("\n###### In build_and_install_workload #####\n")
 
         if test_config_dict['model_name'] == 'bert':
-            self.download_bert_models()
+            self.download_bert_models(test_config_dict)
```
[the remark this answers is not in the document]
Comment (vasanth-intel) [reply baremetal_benchmarking/workloads/tensorflow_workload.py]: Corrected.

File: baremetal_benchmarking/workloads/tensorflow_workload.py [hunk 7]
```diff
@@ -45,29 +45,29 @@ def download_bert_models(self):
...
         elif test_config_dict['model_name'] == 'resnet':
-            self.download_resnet_models()
+            self.download_resnet_models(test_config_dict)
```
[the remark this answers is not in the document]
Comment (vasanth-intel) [reply baremetal_benchmarking/workloads/tensorflow_workload.py]: Corrected.

**Review by jinengandhi-intel — APPROVED** [review]

**Pull request merged**